java-agent/build.gradle.kts

```diff
@@ -178,6 +190,7 @@ kotlin {
             "mu",
         )
         val runtimeJar by registering(ShadowJar::class) {
+            dependsOn(named<KotlinCompile>("compileKotlinJvm"))
```

Comment:
Why search by name is used instead of predefined variables (compileKotlinTask)?

Reply:
compileKotlinTask was deprecated. Let me try to rewrite with `compileTaskProvider`